### Part A – Central Tendency and Spread

### Marks: 50, 60, 65, 70, 75, 80, 85, 90
### Find mean, median, mode.
### Replace 90 with 900 and recompute mean. Explain the effect of outliers.


In [2]:
import numpy as np
import pandas as pd
marks = [50,60,65,70,75,80,85,90]
df = pd.DataFrame({'Marks':marks})

Mean:  71.875


In [3]:
mean = df['Marks'].mean()
print('Mean: ',mean)

Mean:  71.875


In [4]:
median = df['Marks'].median()
print('Median: ',median)

Median:  72.5


In [5]:
mode = df['Marks'].mode()
print('Mode: ',mode)

Mode:  0    50
1    60
2    65
3    70
4    75
5    80
6    85
7    90
Name: Marks, dtype: int64


In [6]:
# Replacing 90 with 900 and calculating mean, median, mode
df['Marks'] = df['Marks'].replace(90,900)
df

Marks
0     50
1     60
2     65
3     70
4     75
5     80
6     85
7    900

In [7]:
mean = df['Marks'].mean()
print('Mean: ',mean)

Mean:  173.125


In [8]:
median = df['Marks'].median()
print('Median: ',median)

Median:  72.5


In [9]:
mode = df['Marks'].mode()
print('Mode: ',mode)

Mode:  0     50
1     60
2     65
3     70
4     75
5     80
6     85
7    900
Name: Marks, dtype: int64


### Explanation of effects of outlier

when there was a value of 90, we got mean as 71.875, median as 72.5 and mode is calculated for categorical values. Since there is no categorical value so mode value is not important. Now when we changed 90 to 900 in dataframe,  we observed mean value is changed as 173.125 and median value changed as 72.5. In both case median value remains close or unchanged. So we can say, median value is outlier resustance but mean value is outlier sensitive.  

### Temperatures (°C): 29, 31, 33, 33, 32, 31, 30
### Calculate variance and standard deviation.
### Which measure is easier to interpret and why?


In [19]:
temperature = [29,31,33,33,32,31,30]
df = pd.DataFrame({'Temperature':temperature})

mean = df['Temperature'].mean()
pop_variance = np.var(df['Temperature'], ddof=0) # population variance
sam_variance = np.var(df['Temperature'], ddof=1) # sample variance
pop_std = np.std(df['Temperature'], ddof= 0) # population SD
sam_std = np.std(df['Temperature'],ddof=1) # sample SD

print('Mean: ',mean)
print('Population Variance: ',pop_variance)
print('Sample Variance: ',sam_variance)
print('Population Standard Deviation: ',pop_std)
print('Sample Standard Deviation: ',sam_std)

Mean:  31.285714285714285
Population Variance:  1.9183673469387752
Sample Variance:  2.2380952380952377
Population Standard Deviation:  1.3850513878332367
Sample Standard Deviation:  1.4960264830861911


standard deviation is easier to interept because it is single unit where variance is in square unit.

### Part B – Percentiles, IQR and Z-Score

### Cat weights (kg): 2.5, 3.0, 3.2, 3.3, 3.4, 3.5, 3.6, 3.9, 4.0, 4.5
### Find P25, P50, P75 and IQR. Mark possible outliers.


In [21]:
weight = [2.5,3.0,3.2,3.3,3.4,3.5,3.6,3.9,4.0,4.5]
#df = pd.DataFrame({'cat_weight': weight })
print('25 Percentile (Q1): ', np.percentile(weight, 25))
print('50 Percentile (Q2): ', np.percentile(weight, 50))
print('75 Percentile (Q3): ', np.percentile(weight, 75))

25 Percentile (Q1):  3.225
50 Percentile (Q2):  3.45
75 Percentile (Q3):  3.825


In [24]:
Q3 = np.percentile(weight, 75)
Q1 = np.percentile(weight, 25)

IQR = Q3-Q1
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR
print('IQR: ',IQR)
print('Outlier Range : ',lower_bound, "To", upper_bound)

IQR:  0.6000000000000001
Outlier Range :  2.325 To 4.7250000000000005


### Given μ = 50, σ = 10
### Find Z for x = 65 and interpret.
### For Z = −2.5, find x.
